src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp

```diff
@@ -277,8 +289,16 @@ static ColumnWithTypeAndName readColumnWithDate32Data(std::shared_ptr<arrow::Chu
             {
                 Int32 days_num = static_cast<Int32>(chunk.Value(value_i));
                 if (days_num > DATE_LUT_MAX_EXTEND_DAY_NUM || days_num < -DAYNUM_OFFSET_EPOCH)
-                    throw Exception{ErrorCodes::VALUE_IS_OUT_OF_RANGE_OF_DATA_TYPE,
-                            "Input value {} of a column \"{}\" is out of allowed Date32 range, which is [{}, {}]", days_num, column_name, DAYNUM_OFFSET_EPOCH, DATE_LUT_MAX_EXTEND_DAY_NUM};
+                {
+                    if (likely(date_time_overflow_mode == DateTimeOverflowMode::SATURATE))
+                        days_num = (days_num < -DAYNUM_OFFSET_EPOCH) ? -DAYNUM_OFFSET_EPOCH : DATE_LUT_MAX_EXTEND_DAY_NUM;
+                    else
+                    {
+                        throw Exception{ErrorCodes::VALUE_IS_OUT_OF_RANGE_OF_DATA_TYPE,
```

Comment:
Begs the question why we either don't throw the new error code `DATE_TIME_OVERFLOW` here or if we go with `VALUE_IS_OUT_OF_RANGE_OF_DATA_TYPE`, why don't we use that consistently everywhere in the code base instead of `DATE_TIME_OVERFLOW`?

Reply:
Agree. I didn't notice that. Moved DATE_TIME_OVERFLOW -> VALUE_IS_OUT_OF_RANGE_OF_DATA_TYPE